test/acceptance/features/steps/openshift.py

```diff
@@ -148,15 +194,24 @@ def expose_service_route(self, service_name, namespace):
         return re.search(r'.*%s\sexposed' % service_name, output)
 
     def get_route_host(self, name, namespace):
-        (output, exit_code) = self.cmd.run(f'oc get route {name} -n {namespace} -o "jsonpath={{.status.ingress[0].host}}"')
-        exit_code | should.be_equal_to(0)
+        output, exit_code = self.cmd.run(f'oc get route {name} -n {namespace} -o "jsonpath={{.status.ingress[0].host}}"')
+        exit_code | should.be_equal_to(0).desc(f"Getting route host exit code should be 0\nERROR: {output}")
         return output
 
-    def check_for_deployment_status(self, deployment_name, namespace, wait_for_status="True"):
-        deployment_status_cmd = f'oc get deployment {deployment_name} -n {namespace} -o "jsonpath={{.status.conditions[*].status}}"'
-        deployment_status, exit_code = self.cmd.run_wait_for_status(deployment_status_cmd, wait_for_status, 5, 300)
-        exit_code | should.be_equal_to(0)
-        return deployment_status
+    def get_deployment_status(self, deployment_name, namespace, wait_for_status=None):
+        deployment_status_cmd = f'oc get deployment {deployment_name} -n {namespace} -o json' \
+            + ' | jq -rc \'.status.conditions[] | select(.type=="Available").status\''
+        output = None
+        exit_code = -1
+        if wait_for_status is not None:
+            status_found, output, exit_code = self.cmd.run_wait_for_status(deployment_status_cmd, wait_for_status, 5, 300)
+            if exit_code == 0:
+                status_found | should.be_truthy.desc(f"Deployment {deployment_name} status should be {wait_for_status},\n ERROR: {output}")
```

Comment:
see above

Reply:
I have replaced should with assert with corresponding error message.